test/plugins/signoff/sagas_test.js

```diff
@@ -5,177 +5,278 @@ import { notifySuccess } from "../../../src/actions/notifications";
 import { routeLoadSuccess } from "../../../src/actions/route";
 import { setClient } from "../../../src/client";
 import * as actions from "../../../src/plugins/signoff/actions";
+import * as collection_actions from "../../../src/actions/collection";
 import * as saga from "../../../src/plugins/signoff/sagas";
 
-describe("signoff plugin sagas", () => {
-  let bucket, collection, getState;
-
-  before(() => {
-    collection = {
-      getData() {},
-      setData() {},
-    };
-    bucket = {
-      collection() {
-        return collection;
-      },
-    };
-    setClient({
-      bucket() {
-        return bucket;
-      },
-    });
-    getState = () => ({
-      bucket: { data: { id: "buck" } },
-      collection: { data: { id: "coll", last_modified: 42 } },
-      signoff: { resource: { destination: { lastSigned: 53 } } },
+describe("Signoff plugin sagas", () => {
+  describe("list hook", () => {
+    let getState;
+
+    describe("onCollectionRecordsRequest()", () => {
+      before(() => {
+        getState = () => ({
+          session: {
+            serverInfo: {
+              capabilities: {
+                signer: {
+                  resources: [
+                    {
+                      source: {
+                        bucket: "stage",
+                        collection: "plugins",
+                      },
+                      preview: {
+                        bucket: "preview",
+                        collection: "plugins",
+                      },
+                      destination: {
+                        bucket: "prod",
+                        collection: "plugins",
```

Comment:
Maybe change these collections so we're sure it isn't matching the second case?

Are we actually going to support these sorts of configurations, with data potentially matching multiple signer configs?

Reply:
> Are we actually going to support these sorts of configurations, with data potentially matching multiple signer configs?

It will take the first one anyway. We can add additional safety checks on the server. https://github.com/Kinto/kinto-signer/issues/216